README.md

```diff
@@ -61,7 +61,7 @@ For more information about spotless checks see
     Example configs/search_config.yml:
     ```yaml
     endpointUrl:
-      productRetrieve: http://localhost:9041/mis/product/
+      datasetRetrieve: http://localhost:9041/dataset
```

Comment:
❗️Update the rest of the repo to use the right terms. E.g.
- `product` should never be used anywhere in the repo
- `datasetId` instead of `id`